### PR TITLE
Refresh CSRF tokens

### PR DESCRIPTION
**Pull request by rauchy**
## What type of PR is this? (check all applicable)
- [x] Feature

## Description
CSRF tokens should expire and reset every once in awhile. This PR expires them every 6 hours.
To prevent a scenario where users keep their Redash tab open for over 6 hours and then can't use the app since they have a stale CSRF token, I've added a retry mechanism in Axios.

* Also turns out Axios does CSRF cookie to header on it's own, so ca3e44a simplifies our code a bit and removes a dependency.

## Related Tickets & Documents
#5055 

## Mobile & Desktop Screenshots/Recordings (if there are UI changes)
image

### CODE REVIEW

File: client/app/services/axios.js [hunk 1]
```diff
@@ -1,26 +1,36 @@
 import axiosLib from "axios";
 import { Auth } from "@/services/auth";
 import qs from "query-string";
-import Cookies from "js-cookie";
+import createAuthRefreshInterceptor from "axios-auth-refresh";
 
 export const axios = axiosLib.create({
   paramsSerializer: params => qs.stringify(params),
+  xsrfCookieName: "csrf_token",
+  xsrfHeaderName: "X-CSRF-TOKEN",
```
Comment (arikfr) [inline client/app/services/axios.js:9]: Worth adding some config around this to disable if needed.

File: client/app/services/axios.js [hunk 2]
```diff
@@ -1,26 +1,36 @@
 import axiosLib from "axios";
 import { Auth } from "@/services/auth";
 import qs from "query-string";
-import Cookies from "js-cookie";
+import createAuthRefreshInterceptor from "axios-auth-refresh";
```
Comment (arikfr) [inline client/app/services/axios.js:4]: @kravets-levko relevant for #5178.

**Review by arikfr — APPROVED** [review]

File: client/app/services/axios.js (hunk 1 above)
Comment (rauchy) [reply client/app/services/axios.js:9]: If the cookie is absent, the header is not added.

https://github.com/axios/axios/blob/master/lib/adapters/xhr.js#L119

File: client/app/services/axios.js [hunk 3]
```diff
@@ -1,26 +1,36 @@
...
+  failedRequest => {
+    const message = failedRequest.response.data.message || "";
+    if (message.includes("CSRF")) {
+      return axios.get("/ping");
+    } else {
+      return Promise.reject();
+    }
+  },
+  { statusCodes: [400] }
+);
```
Comment (arikfr) [inline client/app/services/axios.js:27]: Thinking about it further, I think this whole behavior needs to be somehow configurable. In the Databricks fork this behavior won't work/relevant but we would want to implement a different auth interceptor. 

Does the lib provide a way to remove an interceptor?

Comment (rauchy) [reply client/app/services/axios.js]: Yes, you can easily eject interceptors.

**Review by kravets-levko — APPROVED** [review]

**Pull request merged**